Summoner Heal update.

yasinallahdev opened the pull request:
Summoner: Heal was not applying a debuff that prevented its use from being chained multiple times. Additionally, its movement speed was not attatched to a buff, allowing it to stack short bursts of movement speed when casts were spammed. (Note that particles are still added outside of Heal's buff; this is to be fixed in a later update when a Particle's source is not required to be a champion)

Commit:
Merge pull request #1 from LeagueSandbox/indev

Updated Subrepo.

Commit:
Changed Handling of Summoner Heal.

Heal now uses a Buff Script for the movement speed bonus it grants. Additionally, the HealCheck buff has been added, which lowers the effectiveness of Heal when cast multiple times within 35 seconds.

FurkanS1821 commented on Champions/Global/SummonerHeal.cs:
```diff
@@ -39,41 +39,32 @@ public void OnFinishCasting(Champion owner, Spell spell, AttackableUnit target)
             if (mostWoundedAlliedChampion != null)
             {
-                newHealth = mostWoundedAlliedChampion.Stats.CurrentHealth + 75 + owner.Stats.Level * 15;
+                healthGain = 75 + (owner.Stats.Level * 15);
```
I think this healing routine can be extracted into its own function, so this could be like this:
```cs
public void OnFinishCasting()
{
    // ...
    DoHeal(mostWoundedAlliedChampion);
    DoHeal(owner);
}

public void DoHeal(Champion target)
{
    // heal
}
```

FurkanS1821 reviewed (approved):
Changes seem pretty much okay, except one not-really-necessary-but-would-be-quite-good addition.

Commit:
Updated Heal.

Moved Heal's logic into a PerformHeal function, which calculates the health that would be gained (including checking for the healing reduced debuff) before adding Particles and BuffScripts for Heal's Movement Speed and Healing Debuff.
